torch/ao/quantization/quantizer/xnnpack_quantizer_utils.py

```diff
@@ -318,129 +325,114 @@ def _annotate_conv_relu(
-        assert isinstance(weight, Node)
-        input_qspec_map[weight] = get_weight_qspec(quantization_config)
+    return _do_annotate_conv_bn(gm, quantization_config, filter_fn, _get_pattern)
 
-        bias = conv_node.args[2] if len(conv_node.args) > 2 else None
-        if isinstance(bias, Node):
-            input_qspec_map[bias] = get_bias_qspec(quantization_config)
 
-        conv_node.meta["quantization_annotation"] = QuantizationAnnotation(
-            input_qspec_map=input_qspec_map, _annotated=True
-        )
-
-        bn_output_node.meta["quantization_annotation"] = QuantizationAnnotation(
-            output_qspec=get_output_act_qspec(quantization_config),  # type: ignore[arg-type]
-            _annotated=True,
-        )
-        nodes_to_mark_annotated = list(conv_partition.nodes)
-        nodes_to_mark_annotated.extend(list(bn_partition.nodes))
-        _mark_nodes_as_annotated(nodes_to_mark_annotated)
-    return annotated_partitions
-
-
-@register_annotator("conv2d_bn_relu")
-def _annotate_conv2d_bn_relu(
+@register_annotator("conv_bn_relu")
+def _annotate_conv_bn_relu(
     gm: torch.fx.GraphModule,
     quantization_config: Optional[QuantizationConfig],
     filter_fn: Optional[Callable[[Node], bool]] = None,
 ) -> Optional[List[List[Node]]]:
     """
-    Find Conv2d + batchnorm + relu parititions
+    Find conv + batchnorm + relu parititions
     Note: This is only used for QAT. In PTQ, batchnorm should already be fused into the conv.
     """
-    fused_partitions = find_sequential_partitions(
-        gm, [torch.nn.Conv2d, torch.nn.BatchNorm2d, torch.nn.ReLU], filter_fn
-    )
-    annotated_partitions = []
-    for fused_partition in fused_partitions:
-        conv_partition, bn_partition, relu_partition = fused_partition
-        annotated_partitions.append(
-            conv_partition.nodes + bn_partition.nodes + relu_partition.nodes
-        )
-        if len(relu_partition.output_nodes) > 1:
-            raise ValueError("Relu partition has more than one output node")
-        relu_node = relu_partition.output_nodes[0]
-        if len(conv_partition.output_nodes) > 1:
-            raise ValueError("conv partition has more than one output node")
-        conv_node = conv_partition.output_nodes[0]
-        conv_node_users = list(conv_node.users.keys())
-        if len(conv_node_users) > 1:
-            raise ValueError(
-                "Conv node must be consumed by BN only for it to be fusable."
-            )
-        if len(bn_partition.output_nodes) > 1:
-            raise ValueError("BatchNorm partition has more than one output node")
-        bn_output_node = bn_partition.output_nodes[0]
+    def _get_pattern(conv_fn: Callable):
+        def _conv_bn_relu(x, conv_weight, conv_bias, bn_weight, bn_bias, bn_rm, bn_rv):
+            conv = conv_fn(x, conv_weight, conv_bias)
+            bn = F.batch_norm(conv, bn_rm, bn_rv, bn_weight, bn_bias, training=True)
+            relu = F.relu(bn)
```

Comment:
nit: is it possible to also merge this with the other _get_pattern function as well? e.g. `_get_pattern(conv_fn: Callable, has_relu: bool)`?

Reply:
Interesting. Yeah I can try